conversion.qmd

```diff
@@ -2,3 +2,152 @@
 title: "Conversion to and from R data"
 ---
 
+One of the key goals with extendr, is to provide a framework that allows you to write Rust functions, that interact with R, without having to know the intricacies within R internals, or even R's C-facilities. However, this is unavoidable if one wishes to understand why the extendr-api is the way it is.
+
+Thus, for introducing extendr, we shall mention facts about R internals, but these are not necessary to keep in mind going forward.
+
+```{r, echo=FALSE}
+# to enable extendr/extendrsrc chunks
+library(rextendr)
+# names(knitr::knit_engines$get())
+```
+
+A fundamental data-type in R is the 32-bit integer, `int` in C, and `i32` in Rust. Passing that type around is essential, and straight forward:
```

Comment:
If the target audience for these tutorials is useRs (really need a better name for our community), I think it might make sense to just start with the vector data structure irrespective of data type. Then mention there's this whole thing called "type safety" that R plays fast and loose with and Rust enforces ruthlessly, which makes it a friction point for translating between the languages.

Reply:
Maybe I should have started with this question: "how much background knowledge do we want to assume with these?"